lib/ret/entity_state/create_message.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Ret.EntityState.CreateMessage do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Ret.Hub
+  alias Ret.EntityState.{CreateMessage, UpdateMessage}
+  @schema_prefix "ret0"
+  @primary_key {:entity_create_message_id, :id, autogenerate: true}
```

Comment:
The custom keys we use allow us to shard across multiple data stores, while keeping some desirable properties.

I believe [the code that generates the sharding ids](https://github.com/mozilla/reticulum/blob/6d2d9df5b85a750b974cecb6c2de7fa4ce301ecb/priv/repo/migrations/20170917234137_create_sharding_ids.exs#L9-L23) comes from [this blog post](https://instagram-engineering.com/sharding-ids-at-instagram-1cf5a71e5a5c).

From the blog post:
> The typical solution that works for a single database — just using a database’s natural auto-incrementing primary key feature — no longer works when data is being inserted into many databases at the same time.

...

> Before starting out, we listed out what features were essential in our system:
> 1. Generated IDs should be sortable by time (so a list of photo IDs, for example, could be sorted without fetching more information about the photos)
> 2. IDs should ideally be 64 bits (for smaller indexes, and better storage in systems like Redis)
> 3. The system should introduce as few new ‘moving parts’ as possible — a large part of how we’ve been able to scale Instagram with very few engineers is by choosing simple, easy-to-understand solutions that we trust.

Reply:
I see two counterpoints:
1. We aren’t sharding, so it’s [YAGNI](https://martinfowler.com/bliki/Yagni.html).
2. We are autogenerating the IDs the same as if we used the out-of-the-box behavior.  All this custom configuration (which spans numerous files) is doing is changing the _name_ of the primary ID.

Reply:
Ah right, we are also giving custom _names_ to the primary keys. 

I don't fully understand the reason why this was important. Perhaps something to do with wanting to avoid passing anonymous `id`s around, and then having to track back to figure out why the thing you thought was a `foo.id` was actually a `bar.id`. 

I'm not repeating the pattern because I endorse it. I'm repeating it because it's used for every other table. If we feel strongly that it's bad then we should change it. (But I don't think this PR is the appropriate place to break that pattern.)